--- Python-2.6.8/Include/pyport.h	2012-04-10 17:32:04.000000000 +0200
+++ Python-2.6.8-sim/Include/pyport.h	2012-05-29 16:18:52.000000000 +0200
@@ -129,9 +129,9 @@
 #endif
 
 /* Largest positive value of type Py_ssize_t. */
-#define PY_SSIZE_T_MAX ((Py_ssize_t)(((size_t)-1)>>1))
+#define PY_SSIZE_T_MAX TMP_MAX
 /* Smallest negative value of type Py_ssize_t. */
-#define PY_SSIZE_T_MIN (-PY_SSIZE_T_MAX-1)
+#define PY_SSIZE_T_MIN -TMP_MAX
 
 #if SIZEOF_PID_T > SIZEOF_LONG
 #   error "Python doesn't support sizeof(pid_t) > sizeof(long)"
