--- Python-2.6.8/Modules/posixmodule.c	2012-04-10 17:32:09.000000000 +0200
+++ Python-2.6.8-sim/Modules/posixmodule.c	2012-05-29 16:20:12.000000000 +0200
@@ -361,7 +361,7 @@
 #endif
 
 /* Return a dictionary corresponding to the POSIX environment table */
-#ifdef WITH_NEXT_FRAMEWORK
+#ifdef __APPLE__
 /* On Darwin/MacOSX a shared library or framework has no access to
 ** environ directly, we must obtain it with _NSGetEnviron().
 */
@@ -383,7 +383,7 @@
     d = PyDict_New();
     if (d == NULL)
         return NULL;
-#ifdef WITH_NEXT_FRAMEWORK
+#ifdef __APPLE__
     if (environ == NULL)
         environ = *_NSGetEnviron();
 #endif
